toolkit/tools/downloader/downloader.go

```diff
@@ -96,7 +97,7 @@ func main() {
 		}
 	}
 
-	err = downloader.DownloadFile(*srcUrl, *dstFile, caCerts, tlsCerts)
+	_, err = network.DownloadFileWithRetry(context.Background(), *srcUrl, *dstFile, caCerts, tlsCerts)
```

Comment:
+1 on use of context.

I feel like we should have a context with deadline here in the top level, with a sane max timeout - under which our asssumption should be that the downloader even after retries and all is done. This would put a upper level bounding on our downloader.
What do you folks think ?

Reply:
Not opposed to it... we would probably need to adjust the behavior of the retry to always return an error if it is cancelled though (which isn't the worst idea... currently it returns the error from the last time it did a retry attempt).

Reply:
Ok, gave it a go